src/Data/Map/Internal.purs

```diff
@@ -130,14 +130,32 @@ instance bindMap :: Ord k => Bind (Map k) where
   bind m f = mapMaybeWithKey (\k -> lookup k <<< f) m
 
 instance foldableMap :: Foldable (Map k) where
-  foldl   f z m = foldl   f z (values m)
-  foldr   f z m = foldr   f z (values m)
-  foldMap f   m = foldMap f   (values m)
+  foldr f z m = case m of
+    Leaf -> z
+    Two ml _ v mr -> foldr f (f v (foldr f z mr)) ml
+    Three ml _ v1 mm _ v2 mr -> foldr f (f v1 (foldr f (f v2 (foldr f z mr)) mm)) ml
+  foldl f z m = case m of
+    Leaf -> z
+    Two ml _ v mr -> foldl f (foldl f (f z v) ml) mr
```

Comment:
I think this is wrong. Shouldn't it be
```diff
-Two ml _ v mr -> foldl f (foldl f (f z v) ml) mr
+Two ml _ v mr -> foldl f (f v (foldl f z ml)) mr
```

Reply:
why do u think that it should be so?

Reply:
your variant looks like as fold center

Reply:
```
mapFoldl :: forall k a b. (b -> a -> b) -> b -> Map k a -> b
mapFoldl f z m = case m of
  Leaf -> z
  Two ml _ v mr -> mapFoldl f (mapFoldl f (f z v) ml) mr
  Three ml _ v1 mm _ v2 mr -> mapFoldl f (mapFoldl f (f (mapFoldl f (f z v1) ml) v2) mm) mr

mapFoldl' :: forall k a. (a -> a -> a) -> a -> Map k a -> a
mapFoldl' f z m = case m of
  Leaf -> z
  Two ml _ v mr -> mapFoldl' f (f v (mapFoldl' f z ml)) mr
  Three ml _ v1 mm _ v2 mr -> mapFoldl' f (f (mapFoldl' f (f (mapFoldl' f z ml) v1) mm) v2) mr
  ```
  
  `mapFoldl` my variant, `mapFoldl'` - your

Reply:
```
  log $ mapFoldl (\x y -> "(" <> x <> "-" <> show y <> ")") "0" m
  log $ mapFoldl' (\x y -> "(" <> x <> "-" <> show y <> ")") "0" m
  ```
  
  ```
((((((((((0-4)-2)-1)-3)-6)-5)-8)-7)-9)-10)
((((((((((0-1)-2)-3)-4)-5)-6)-7)-8)-9)-10)
```